### PR TITLE
Update C3 OpenAPI template to allow users to pick either Hono or itty-router

### PR DESCRIPTION
**Pull request by G4brym**
## What this PR solves / how to test

This pr updates the C3 OpenAPI template to allow users to pick either Hono or itty-router.

Example usage:
`npm create cloudflare@latest -- --type openapi --framework hono`
`npm create cloudflare@latest -- --type openapi --framework itty-router`

## Author has addressed the following

- Tests
  - [ ] TODO (before merge)
  - [ ] Included
  - [x] Not necessary because:
- E2E Tests CI Job required? (Use "e2e" label or ask maintainer to run separately)
  - [ ] I don't know
  - [ ] Required / Maybe required
  - [x] Not required because:
- Changeset ([Changeset guidelines](https://github.com/cloudflare/workers-sdk/blob/main/CONTRIBUTING.md#changesets))
  - [ ] TODO (before merge)
  - [x] Included
  - [ ] Not necessary because:
- Public documentation
  - [ ] TODO (before merge)
  - [ ] Cloudflare docs PR(s): 
  - [x] Not necessary because:

### CODE REVIEW

File: .changeset/metal-seahorses-occur.md [hunk 1]
```diff
@@ -0,0 +1,6 @@
+---
+"create-cloudflare": minor
+"@cloudflare/prerelease-registry": patch
+---
+
+Update c3 openapi template to use chanfana and Hono
```
Comment (CarmenPopoviciu) [inline .changeset/metal-seahorses-occur.md:6]:
```suggestion
fix: Update c3 openapi template to use chanfana and Hono
```

**Review by jculvey — APPROVED** [review]

**Comment by G4brym** [comment]
Builds are failing because the PR is done from a forked repo, but should otherwise be green

**Comment by G4brym** [comment]
I've updated the pull request, to allow users to pick either Hono or itty-router as the base framework for this c3 template.
Pull request is ready to be merged

File: packages/create-cloudflare/templates/openapi/hono/package.json [hunk 2]
```diff
@@ -3,15 +3,18 @@
...
 	"scripts": {
+		"cf-typegen": "wrangler types",
 		"deploy": "wrangler deploy",
 		"dev": "wrangler dev",
-		"start": "wrangler dev",
-		"cf-typegen": "wrangler types"
+		"start": "wrangler dev"
 	},
 	"dependencies": {
-		"@cloudflare/itty-router-openapi": "^1.0.1"
```
Comment (marceloverdijk) [inline packages/create-cloudflare/templates/openapi/hono/package.json:12]: Is this really needed?

Comment (G4brym) [reply packages/create-cloudflare/templates/openapi/hono/package.json:12]: yes, this old dependency needs to be removed

**Comment by G4brym** [comment]
Hey @jculvey , the C3 E2E builds are failing for reasons not related to this pr, do you think you could merge this?

**Review by petebacondarwin — APPROVED** [review]
I pushed this branch to our fork to run the e2e tests. See https://github.com/cloudflare/workers-sdk/pull/6207.
They are now green.

**Pull request merged**